### PR TITLE
Issue1176 improve ios marker performance by X100

### PR DESCRIPTION
**Pull request by ericapply**
Fix problem https://github.com/airbnb/react-native-maps/issues/1176

When image props is provided, it will re-use UIImage if the image is same.

This will increase the performance by more than X100

### CODE REVIEW

**Comment by ryancoughlin** [comment]
Bump -- would love to get this in. Running into the same issue.

**Comment by christopherdro** [comment]
Looks great.
Can we get the merge conflicts resolved so we can give this a proper review?

**Comment by ericapply** [comment]
Really busy today, I will try to resolve merge conflicts asap

**Comment by ryancoughlin** [comment]
Thanks @ericapply, been using a forked version of this to get past this. Huge improvement.

**Comment by bosung90** [comment]
Please merge!🎉

**Comment by grundmanise** [comment]
@christopherdro please merge this PR

File: lib/ios/AirGoogleMaps/GlobalVars.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+//  GlobalVars.h
+//  EvoApp
```
Comment (christopherdro) [inline lib/ios/AirGoogleMaps/GlobalVars.h]: Remove reference to EvoApp

**Comment by christopherdro** [comment]
@spikebrehm @lelandrichardson  Any chance you can take a look?

This PR still needs to be tested and reviewed. There is no sense is rushing to merge this.
@bosung90 @grundmanise You can do what @ryancoughlin does for the time being if you really need this feature.

**Comment by ryancoughlin** [comment]
Bless your heart @christopherdro 🙌 🙏 

I was getting nervous.

**Comment by christopherdro** [comment]
This has been reverted until we can resolve https://github.com/airbnb/react-native-maps/issues/1560

**Comment by ngandhy** [comment]
@ericapply - any luck solving #1560?

Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_GlobalVars", referenced from:
      objc-class-ref in AIRGoogleMapMarker.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by grubstarstar** [comment]
@ericapply @christopherdro Any luck getting this going? I'm desperately in need of getting this if there's a workaround or a patch or anything? Thanks!

**Pull request merged**